modules/algebrasAndModel/src/main/scala/kafkaAlgebra/KafkaAlgebra.scala

```diff
@@ -0,0 +1,24 @@
+package kafkaAlgebra
+
+import cats.effect.Sync
+import fs2.kafka.{ProducerRecords, ProducerSettings}
+import fs2.kafka.vulcan.{Auth, AvroSettings, SchemaRegistryClientSettings}
+import utils.Types.{Password, SchemaRegistryUrl, UserName}
+import fs2.Stream
+
+trait KafkaAlgebra[F[_], K, V] {
+
+  def publish(key: K, messageEvent: V): Stream[F, Unit]
+  def producerSettings: ProducerSettings[F, K, V]
+  def avroSettings(
```

Comment:
Same as above, plus, in tagless final, we don't have concrete methods.

Reply:
sweet, thank you Fede.
I will amend this